thewalrus/decompositions.py

```diff
@@ -100,75 +320,3 @@ def williamson(V, rtol=1e-05, atol=1e-08):
     Db = np.diag(dd + dd)
     S = Mm12 @ Ktt @ sqrtm(Db)
     return Db, np.linalg.inv(S).T
-
-
-def symplectic_eigenvals(cov):
-    r"""Returns the symplectic eigenvalues of a covariance matrix.
-
-    Args:
-        cov (array): a covariance matrix
-
-    Returns:
-        (array): symplectic eigenvalues
-    """
-    M = int(len(cov) / 2)
-    D, _ = williamson(cov)
-    return np.diag(D)[:M]
-
-
-def blochmessiah(S):
-    """Returns the Bloch-Messiah decomposition of a symplectic matrix S = uff @ dff @ vff
-       where uff and vff are orthogonal symplectic matrices and dff is a diagonal matrix
-       of the form diag(d1,d2,...,dn,d1^-1, d2^-1,...,dn^-1),
-
-    Args:
-        S (array[float]): 2N x 2N real symplectic matrix
-
-    Returns:
-        tupple(array[float],  : orthogonal symplectic matrix uff
-               array[float],  : diagional matrix dff
-               array[float])  : orthogonal symplectic matrix vff
-    """
-
-    N, _ = S.shape
-
-    if not is_symplectic(S):
-        raise ValueError("Input matrix is not symplectic.")
-
-    # Changing Basis
-    R = (1 / np.sqrt(2)) * np.block(
-        [[np.eye(N // 2), 1j * np.eye(N // 2)], [np.eye(N // 2), -1j * np.eye(N // 2)]]
-    )
-    Sc = R @ S @ np.conjugate(R).T
-    # Polar Decomposition
-    u1, d1, v1 = np.linalg.svd(Sc)
-    Sig = u1 @ np.diag(d1) @ np.conjugate(u1).T
-    Unitary = u1 @ v1
-    # Blocks of Unitary and Hermitian symplectics
-    alpha = Unitary[0 : N // 2, 0 : N // 2]
-    beta = Sig[0 : N // 2, N // 2 : N]
-    # Bloch-Messiah in this Basis
-    u2, d2, v2 = np.linalg.svd(beta)
-    sval = np.arcsinh(d2)
-    takagibeta = u2 @ sqrtm(np.conjugate(u2).T @ (v2.T))
-    uf = np.block([[takagibeta, 0 * takagibeta], [0 * takagibeta, np.conjugate(takagibeta)]])
-    vf = np.block(
-        [
-            [np.conjugate(takagibeta).T @ alpha, 0 * takagibeta],
-            [0 * takagibeta, np.conjugate(np.conjugate(takagibeta).T @ alpha)],
-        ]
-    )
-    df = np.block(
-        [
-            [np.diag(np.cosh(sval)), np.diag(np.sinh(sval))],
-            [np.diag(np.sinh(sval)), np.diag(np.cosh(sval))],
-        ]
-    )
-    # Rotating Back to Original Basis
-    uff = np.conjugate(R).T @ uf @ R
-    vff = np.conjugate(R).T @ vf @ R
-    dff = np.conjugate(R).T @ df @ R
-    dff = np.real_if_close(dff)
-    vff = np.real_if_close(vff)
-    uff = np.real_if_close(uff)
-    return uff, dff, vff
```

Comment:
This should not be deleted

Reply:
It hasn't been deleted, when I moved `autonne` here and added `takagi` I moved the functions such that they're in alphabetical order. All is still there, just reordered.

Reply:
Was that necessary?

Reply:
I saw that some other scripts had been orgnised in that way, so when I moved the functions I had the choice to just put them underneath or organise them in that way. I chose the latter.